### DISCUSSION TITLE
Hub firmware update in offline mode?

### DISCUSSION THREAD

**Discussion by dfigus** [reinhard-brandstaedter/solarflow-control#299] [Q&A] [answered]
I'm running by Hub2k in offline mode with HA and recently a new firmware version was released. Is there a way to trigger the FW update via sf-control or bt-manager? Or do I need to reconnect it to the Zendure cloud and trigger the update via app?

If there is a way to trigger it locally, I could also look into how to wrap this into HA update entities and create a PR/wrap it into my HA add-on ([dfigus/addon-solarflow-control](https://github.com/dfigus/addon-solarflow-control/)).

**Comment by reinhard-brandstaedter** [comment]
I usually update my firmware via the app and a bluetooth connection.
Theoretically you can push a firmware file via MQTT too, but I rather use the app approach.

**Reply by dfigus** [reply]
Is the bluetooth app connection enough or do I need to first re-connect it to the Zendure cloud?

**Reply by reinhard-brandstaedter** [reply, marked as answer]
I usually just walk over to my hub, connect the app via BT (It won't connect via cloud as the hub is offline), update and that's it.
No need to take the hub online again.
If your app doesn't connect via BT, restart the hub and try the BT connection then, but usually BT is active on the hub all the time and takes precedence over WiFi it seems.
